x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/mapper/GeoShapeWithDocValuesFieldMapper.java

```diff
@@ -401,6 +401,16 @@ public GeoBoundingBox getInternalBoundingBox() {
             return boundingBox;
         }
 
+        @Override
+        public GeoPoint getInternalLabelPosition() {
+            try {
+                return value.labelPosition();
+            } catch (IOException e) {
+                // Most of the time the centroid is a good compromise, but perhaps we should know about this error anyway?
+                return centroid;
+            }
```

Comment:
Not sure if we should ignore the error, maybe throw an UncheckedIOException instead?